connect/runtime/src/main/java/org/apache/kafka/connect/runtime/ConnectorConfig.java

```diff
@@ -257,12 +273,25 @@ public boolean includeRecordDetailsInErrorLog() {
         final List<Transformation<R>> transformations = new ArrayList<>(transformAliases.size());
         for (String alias : transformAliases) {
             final String prefix = TRANSFORMS_CONFIG + "." + alias + ".";
+
             try {
                 @SuppressWarnings("unchecked")
                 final Transformation<R> transformation = getClass(prefix + "type").asSubclass(Transformation.class)
                         .getDeclaredConstructor().newInstance();
-                transformation.configure(originalsWithPrefix(prefix));
-                transformations.add(transformation);
+                Map<String, Object> configs = originalsWithPrefix(prefix);
+                Object predicateAlias = configs.remove("predicate");
+                Object negate = configs.remove("negate");
+                transformation.configure(configs);
+                if (predicateAlias != null) {
+                    String predicatePrefix = "predicates." + predicateAlias + ".";
+                    @SuppressWarnings("unchecked")
+                    Predicate<R> predicate = getClass(predicatePrefix + "type").asSubclass(Predicate.class)
+                            .getDeclaredConstructor().newInstance();
```

Comment:
Blegh, was hoping we might be able to use `Utils::newInstance` or `AbstractConfig::newConfiguredInstance` but it looks like neither quite does what we need; the former doesn't do casting to a subclass unless you give it the FQCN of a class instead of an already-loaded `Class<?>` object, and the latter doesn't give enough control over exactly which properties the new instance is configured with.

Since we're using this same logic in several different places in this file alone, we might consider expanding the `Utils` class with a new utility method that does this for us. Maybe something like:

```java
public static Class<T> newInstance(Class<?> klass, Class<T> baseClass) {
    // Return an instance of klass that has been automatically cast to the type of baseClass
}
```